core/lib/Cardano/Address/Style/Shelley.hs

```diff
@@ -962,8 +898,8 @@ unsafeFromRight =
 --- | Computes a 28-byte Blake2b224 digest of a Shelley 'XPub'.
 ---
 --- @since 3.0.0
-hashKey :: Shelley key XPub -> KeyHash
-hashKey = KeyHash . hashCredential . xpubPublicKey . getKey
+hashKey :: KeyType -> Shelley key XPub -> KeyHash
+hashKey cred = flip KeyHash cred . hashCredential . xpubPublicKey . getKey
```

Comment:
Does this actually make sense for Shelley keys? The `KeyHash` data-type seems very much tight to monetary scripts (and when serialized, will assume `_shared` prefixes which is probably not what we want. So either the `KeyHash` should be a little more generic (and also, moved out of the `Script` module), or, we should use a different data-type for this. 

We could likely use a phantom type to capture the purpose on the KeyHash. Or simply a data-type proper at the value-level.

Reply:
it is not needed in Shelley, only in Shared